### DISCUSSION TITLE
hugo-installer usable behind corporate proxy?

### DISCUSSION THREAD

**Discussion by OCram85** [itsdevdom/hugo-installer#37] [Q&A] [answered]
The hugo theme [doks](https://github.com/h-enk/doks) uses the hugo-installer since the latest release.
I'm now having problems with `npm install` where the hugo-installer tries to download  the binaries from github.

Does anyone use the hugo-installer behind a corporate proxy?

![image](https://user-images.githubusercontent.com/13993700/126482878-29b710d7-58d9-4313-95f5-c44098c805aa.png)

I've already tried to set the `http_proxy` and `https_proxy` environment variables. I also use the `npm config set proxy` and `npm config set https-proxy` config command.

If I got this right, hugo-installer uses the [got](https://www.npmjs.com/package/got) package for http request. Does got support using a proxy?

**Comment by itsdevdom** [comment, marked as answer]
I could add a new `--http-proxy [...]` CLI argument or accept an `HTTP_PROXY` environment variable, implemented based on [hpagent](https://github.com/delvedor/hpagent) similar to [here](https://github.com/twilio-labs/serverless-toolkit/pull/228/files#diff-5534af852130a7fea120a9396d11b2efe32ae35ece002e5cac54a96b22d8caeaR93), and publish it as a new feature release.

"doks" is just a template, so am I right assuming you can upgrade hugo-installer and extend the CLI arguments at any given time?

**Reply by OCram85** [reply]
You're right, I could test it locally with a cloned and customized [package.json](https://github.com/h-enk/doks/blob/master/package.json#L33) for doks.

**Reply by OCram85** [reply]
@dominique-mueller: Unfortunately I'm not familiar with Typescript and the used packages. Otherwise I could help with a raw implementation or testing. I couldn't implement the custom got `options` or `client` object with the proxy config.

**Reply by itsdevdom** [reply]
All good, I already did an implementation yesterday but I'm still refining it and trying to test it manually, should be done on the weekends ;)

**Reply by itsdevdom** [reply]
Ok, well, I had some free time during work waiting for builds and used it to finish the feature up. It's published as version `3.1.0` (https://github.com/dominique-mueller/hugo-installer/releases/tag/3.1.0).

So if you intend to download via GitHub (which is the default), you need to use the https proxy configuration, either

- via the `--httpsProxy [url]` CLI argument or
- the `HTTPS_PROXY` environment variable.

Usually, http proxy and https proxy configurations are identical, and usually both start with `http://` (even the https one), but if npm allows us to configure both separately so does hugo-installer now.

Let me know if it works, I really hope it does cause this stuff is really hard to test ;D

**Reply by OCram85** [reply]
Awesome! Thanks for this super fast response and fix. I'll test it on monday when I'm back in the office....

**Comment by OCram85** [comment]
`hugo-installer >= 3.1.0` works perfect with proxy config!

**Reply by itsdevdom** [reply]
Awesome!!
